tests/sdk/java/await_test.go

```diff
@@ -106,23 +116,207 @@ func TestAwaitSkip(t *testing.T) {
+	} `json:"spec"`
+	Status struct {
+		Conditions []struct {
+			Status string `json:"status"`
+			Type   string `json:"type"`
+		} `json:"conditions"`
+	} `json:"status"`
+}
+
+type expectation struct {
+	name            string
+	someField       string
+	conditionType   string
+	conditionStatus string
+}
+
+func TestAwaitGeneric(t *testing.T) {
+	// No t.Parallel because this touches environment variables.
+
+	waitForCRDs := func(events <-chan events.EngineEvent) {
+		// Wait until we see a resource with a "Waiting for readiness message"
+		t.Helper()
+		t.Log("waiting for a resource to start awaiting")
+		for e := range events {
+			if e.DiagnosticEvent == nil {
+				continue
+			}
+			if strings.Contains(e.DiagnosticEvent.Message, "Waiting for readiness") {
+				go func() {
+					for range events {
+						// Need to exhaust the channel otherwise things deadlock.
+					}
+				}()
+				break
+			}
+		}
+		// Wait an extra second to let any other resources to get applied.
+		time.Sleep(1 * time.Second)
+	}
+
+	touch := func(t *testing.T, dir string) {
+		t.Helper()
+		t.Log("touching resources")
+		cmd := exec.Command(filepath.Join(dir, "make-progressing.sh"))
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err := cmd.Run()
+		require.NoError(t, err)
+	}
+
+	makeReady := func(t *testing.T, dir string) {
+		touch(t, dir)
+		time.Sleep(1 * time.Second)
+		t.Log("marking resources ready")
+		t.Helper()
+		cmd := exec.Command(filepath.Join(dir, "make-ready.sh"))
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err := cmd.Run()
+		require.NoError(t, err)
+	}
+
+	assertExpectations := func(t *testing.T, outputs auto.OutputMap, expect []expectation) {
+		t.Helper()
+		for _, want := range expect {
+			obj := outputs[want.name].Value
+			bytes, err := json.Marshal(obj)
+			require.NoError(t, err)
+
+			var resource awaitedResource
+			err = json.Unmarshal(bytes, &resource)
+			require.NoError(t, err)
+
+			assert.Equal(t, want.someField, resource.Spec.SomeField, want.name)
+			if want.conditionType != "" {
+				require.Len(t, resource.Status.Conditions, 1, want.name)
+				assert.Equal(t, want.conditionType, resource.Status.Conditions[0].Type, want.name)
+				assert.Equal(t, want.conditionStatus, resource.Status.Conditions[0].Status, want.name)
+			}
+		}
+	}
+
+	assertReady := func(t *testing.T, outputs auto.OutputMap) {
+		t.Helper()
+		expect := []expectation{
+			{
+				name:            "wantsReady",
+				someField:       "touched",
+				conditionType:   "Ready",
+				conditionStatus: "True",
+			},
+		}
+		assertExpectations(t, outputs, expect)
+	}
+
+	assertUntouched := func(t *testing.T, outputs auto.OutputMap) {
+		t.Helper()
+		expect := []expectation{
+			{
+				name:            "wantsReady",
+				someField:       "untouched",
+				conditionType:   "Ready",
+				conditionStatus: "False",
+			},
+		}
+		assertExpectations(t, outputs, expect)
+	}
+
+	t.Run("enabled", func(t *testing.T) {
+		t.Setenv("PULUMI_K8S_AWAIT_ALL", "true")
```

Comment:
Should we also set PULUMI_EXPERIMENTAL?

Reply:
Discussed and decided to stick with `PULUMI_K8S_AWAIT_ALL`.